packages/chart/src/FigureChartModel.ts

```diff
@@ -180,6 +180,14 @@ class FigureChartModel extends ChartModel {
       this.layout.hiddenlabels = ChartUtils.getHiddenLabels(this.settings);
     }
 
+    // We only want to force hide the legend if there is only one series that is not a PIE
+    // Right now this means that if the user only has one series, they cannot explicitly show the legend until deephaven-core#3254 is implemented
+    // TODO: deephaven-core#3254, once done, this can be removed.
+    this.layout.showlegend =
+      this.data.length > 1 || series.plotStyle === dh.plot.SeriesPlotStyle.PIE
+        ? undefined
+        : false;
```

Comment:
Should this be something like this instead?
`this.layout.showlegend = this.data.length === 1 ? false : showLegend`

Since `showLegend` defaults to true, just use it unless only 1 series. That way a user could disable on pie still if they wanted to?

Reply:
There are two different `showlegend` properties - this one on the entire layout, and then there's one on the data trace itself.
In the JS API it is set on the `JsChart`, which I'm using to set on the all the series below that. In our API, a `Figure` can have multiple `Chart`s, which can have multiple `Series`s. Mapping that to plotly, the `Figure` essentially maps to `layout`, and `Series` maps to trace, whereas `Chart` maps to the axes/properties within a data trace.

And I guess this is a convoluted was of saying yea I think you're right. But we should still remove it once https://github.com/deephaven/deephaven-core/issues/3254 is implemented.